x/concentrated-liquidity/tick.go

```diff
@@ -36,19 +36,6 @@ func (k Keeper) initOrUpdateTick(ctx sdk.Context, poolId uint64, currentTick int
 	// calculate liquidityGross, which does not care about whether liquidityIn is positive or negative
 	liquidityBefore := tickInfo.LiquidityGross
 
-	// if given tickIndex is LTE to the current tick and the liquidityBefore is zero,
-	// set the tick's spread reward growth opposite direction of last traversal to the spread factor accumulator's value
-	if liquidityBefore.IsZero() {
-		if tickIndex <= currentTick {
-			accum, err := k.GetSpreadRewardAccumulator(ctx, poolId)
```

Comment:
Removing these reads should change the gas for this function so this is not state compatible right? Since even if this branch didn't do anything the reads were still happening

Reply:
Right. I will separate it into a separate PR

Reply:
Moved here: https://github.com/osmosis-labs/osmosis/pull/7625